app/services/load_profiles/hierarchy.rb

```diff
@@ -0,0 +1,44 @@
+module LoadProfiles
+  class Hierarchy
+    #
+    # Query to sort Load Profile Categories hierarchly
+    #
+
+    def initialize
+    end
+
+    def tree_sort
+      with_paths.sort do |category_a, category_b|
+        category_a[:path] <=> category_b[:path]
```

Comment:
Fixed it a little differently due to the fact that the paths are all unique. See line 32.